src/Stripe.net/Infrastructure/Public/StripeStreamedResponse.cs

```diff
@@ -0,0 +1,51 @@
+namespace Stripe
+{
+    using System.IO;
+    using System.Net;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// A streaming binary response. Body is not assumed to be text for successful responses.
+    /// </summary>
+    public class StripeStreamedResponse : StripeResponseBase
+    {
+        private Task<StripeResponse> fetchFullyTask;
+
+        public StripeStreamedResponse(HttpStatusCode statusCode, HttpResponseHeaders headers, Stream body)
+            : base(statusCode, headers)
+        {
+            this.Body = body;
+        }
+
+        /// <summary>
+        /// Binary response body as a <see cref="Stream"/>.
+        /// </summary>
+        public Stream Body { get; }
+
+        /// <summary>
+        /// Converts this response into a regular <see cref="StripeResponse"/> by
+        /// reading the body in full.
+        /// </summary>
+        /// This assumes that the response body is textual, which will be the case for
+        /// most API responses and for errors.
+        /// <returns>The response with body fully read.</returns>
+        public async Task<StripeResponse> ToStripeResponseAsync()
+        {
+            // We keep a reference to the task because we can only read the body once.
+            if (this.fetchFullyTask == null)
```

Comment:
Hmm, I didn't think of this in [stripe-java](https://github.com/stripe/stripe-java/pull/1207/files#diff-f748ba076a0736ad824fca475e0594a7d8d118fd6d241ad0a0e047db81e43e49R25). Do you think it's important permit multiple calls there, too?